hdr/sqlite_modern_cpp.h

```diff
@@ -318,6 +323,18 @@ namespace sqlite {
 		database(std::shared_ptr<sqlite3> db):
 			_db(db) {}
 
+		database(const std::string &db_name, const sqlite_config &config):
+			database(std::u16string(db_name.begin(), db_name.end()), config) {}
+
+		database(const std::u16string &db_name, const sqlite_config &config): database(db_name) {
+#ifdef SQLITE_HAS_CODEC
+			if(!config.key.empty()) set_key(config.key);
+#else
+			assert(config.key.empty() && "Encryption supported is disabled.");
```

Comment:
We could use an exception instead, but passing a key without encryption support is IMO a strong indication, that the application is compiled with the wrong settings.